build_scripts/deploy_production.sh

```diff
@@ -1,31 +1,57 @@
+#!/bin/bash
+
 readonly DISTRIBUTION_ID="E2VZR8TOC61OQY"
-readonly MIN_FILE="keen-tracking.min.js"
-
-while true; do
-    read -p "Deploying production. Confirm: (Yes/No) " yn
-    case $yn in
-        [Yy]* ) break;;
-        [Nn]* ) echo "Exiting without deploy..."; exit;;
-        * ) echo "Please answer yes or no.";;
-    esac
+readonly PACKAGE_VERSION=$(node -pe "require('./package.json').version" | tr '[.]' '_')
+readonly FILE_NAME="keen-tracking"
+readonly MINIFIED_FILE="${FILE_NAME}.min.js"
+readonly BACKUP_FILE="${FILE_NAME}_last.min.js"
+readonly VERSIONED_FILE="${FILE_NAME}_${PACKAGE_VERSION}.min.js"
+readonly S3_BUCKET="prodperfect-keen-js"
+
+for arg in "$@"
+do
+    if [ "$arg" == "--force" ] || [ "$arg" == "-f" ]
+    then
+        FORCE_DEPLOY="yes"
+        echo "Using --force option. Will not check if PR has been approved (non-interactive mode)."
+    fi
 done
 
+if [ "$FORCE_DEPLOY" != "yes" ]
+then
+  while true; do
+      read -p "Have you merged your approved pr with tests passing into master? (Yes/No) " yn
+      case $yn in
+          [Yy]* ) break;;
+          [Nn]* ) echo "Please do so before continuing"; exit;;
+          * ) echo "Please answer yes or no.";;
+      esac
+  done
+fi
+
+echo "updating local master";
+git checkout master || { echo 'git checkout master failed!' ; exit 1; }
+git pull origin master || { echo 'git pull origin master failed!' ; exit 1; }
+
 if aws --version &> /dev/null
 then
-  echo "copying current min file to last";
-  aws s3 cp "s3://prodperfect-keen-js/${MIN_FILE}" "s3://prodperfect-keen-js/keen-tracking_last.min.js" --region us-east-2 --acl public-read
+  echo "uploading a versioned build of ${FILE_NAME}.min to s3://${S3_BUCKET}/${VERSIONED_FILE}";
+  aws s3 cp "./dist/${MINIFIED_FILE}" "s3://${S3_BUCKET}/${VERSIONED_FILE}" --region us-east-2 --acl public-read
+ 
+  echo "Making backup of existing '${MINIFIED_FILE}' file.";
+  aws s3 cp "s3://${S3_BUCKET}/${MINIFIED_FILE}" "s3://${S3_BUCKET}/${BACKUP_FILE}" --region us-east-1 --acl public-read
+
+  echo "Uploading latest build of '${MINIFIED_FILE}'."  
+  aws s3 cp "./dist/${MINIFIED_FILE}" "s3://${S3_BUCKET}/${MINIFIED_FILE}" --region us-east-1 --acl public-read
```

Comment:
some files are put on us-east-1, another is on us-east-2. is this correct? thoughts on making an aws region variable at the top?

Reply:
Yeah I wasn't sure what we wanted here, I just followed on from what was there previously.  @pevner-p2 do you have feelings about what region this should be?

Reply:
I'd prefer, for my sanity, that everything were inside us-east-1, if possible.